clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/agent/KubernetesV2CachingAgent.java

```diff
@@ -193,4 +193,10 @@ protected CacheResult buildCacheResult(Map<KubernetesKind, List<KubernetesManife
             });
     return result;
   }
+
+  protected ImmutableList<String> getNamespaces() {
+    return credentials.getDeclaredNamespaces().stream()
+        .filter(n -> agentCount == 1 || Math.abs(n.hashCode() % agentCount) == agentIndex)
```

Comment:
What exactly is this doing?

Reply:
It's handling splitting of namespaces among the caching agents.

So if you have 5 caching agents and this is caching agent 2, this agent only handles namespaces where `Math.abs(n.hashCode() % 5) == 2`.  (If there's only one agent, we need to handle all namespaces.)

The hashing makes sure the assignment of namespaces to caching agents is stable (and not dependent on the order we get the list back from the cluster).